### ISSUE TITLE
Pipeline facade float32 overflow hardening and int32 bounds on dimensions and smoke probes

### ISSUE THREAD

**Issue by deepanshu-yd** [SlopDotCash/asi#437]
## Problem

End-to-end pipeline component configs (`Step2FeatureConfig`, `Step2UPGDConfig`, `Step2AssociativePipelineConfig`, `HordeActorCriticPipelineConfig`) and `run_pipeline_smoke` lack float32 overflow validation on scalar hyperparameters (`ema_decay`, `periods`, `step_size`, `sparsity`, `write_lr`, `retention`, `utility_lr`, `utility_decay`, `min_weight`, `max_weight`, `logit_scale`, `scope_lr`, `budget_lr`, `initial_budget_fraction`, `scope_logit_clip`, `actor_step_size`, `actor_lamda`, `temperature`, `actor_obgd_kappa`), allow adversarial float subclasses with out-of-domain exact ratios, and lack signed int32 bounds on `observation_dim`, `n_heads`, `hidden_sizes`, `vocab_size`, `block_size`, `suffix_length`, `max_features`, `min_effective_budget`, `n_actions`, `value_head_index`, and `run_pipeline_smoke` arguments (`steps`, `seed`).

## Reproduction

```python
from alberta_framework.pipeline import Step2UPGDConfig

# Accepts float32 overflow on step_size
cfg = Step2UPGDConfig(step_size=1e100)
```

## Proposed Fix

- Use `round_real_to_float32_with_ratio` in `alberta_framework/pipeline.py` checking exact integer ratios, domain bounds, and rejecting class-property spoofing.
- Bound integer dimensions, counts, sizes, and smoke arguments to signed int32 bounds (`2**31 - 1`).
- Add comprehensive test coverage in `tests/test_pipeline.py`.
